overlay/generic/smartdc/lib/smartos_prompt_config.sh

```diff
@@ -751,18 +751,16 @@ promptpool()
 
 create_dump()
 {
-	# Get avail zpool size - this assumes we're not using any space yet.
-	base_size=`zfs get -H -p -o value available ${SYS_ZPOOL}`
-	# Convert to MB
-	base_size=`expr $base_size / 1000000`
-	# Calculate 5% of that
-	base_size=`expr $base_size / 20`
-	# Cap it at 4GB
-	[ ${base_size} -gt 4096 ] && base_size=4096
-
-	# Create the dump zvol
-	zfs create -V ${base_size}mb ${SYS_ZPOOL}/dump || \
-	    fatal "failed to create the dump zvol"
+	# Get the estimated dump size from dumpadm -e in mb double it and convert to int
+
+	local dumpsize=$(echo "(( $(dumpadm -e |  sed "s/[^0-9|.]//g") * 1024) * 2 )/1" |bc)
+
+	# Ensure the dumpvol is at least 4G. This will store most kernel page only dumps 
```

Comment:
The 4G was changed to 1G after testing showed that 4G was a waste. The current version uses 1G as the smallest size. Which should be sufficient for a starting point for dumps.